### ISSUE TITLE
Unable to query tables named `version` in another database without escaping

### ISSUE THREAD

**Issue by theme1256** [dolthub/dolt#7237]
I getting the error message `syntax error at position 27 near 'version'` when running this query:
```sql
SELECT * FROM base.version;
```

If I run it like this, is works just fine:
```sql
SELECT * FROM base.`version`;
```

Altso, this works:
```sql
DROP VIEW IF EXISTS version;
CREATE VIEW version AS SELECT * FROM base.`version`;
```

But, this doesn't:
```sql
DROP VIEW IF EXISTS version;
CREATE VIEW version AS SELECT * FROM base.version;
```

**Comment by fulghum** [comment]
Hi @theme1256, thank you for taking the time to report this! I'm glad you found a temporary workaround by using backtick/identifier quotes. 

I confirmed that `version` doesn't require identifier quoting in MySQL, so we'll get this fixed in Dolt's parser so we can match MySQL's behavior.

**Comment by fulghum** [comment]
We just released this fix in [Dolt 1.30.5](https://github.com/dolthub/dolt/releases/tag/v1.30.5) 🎉 

Thanks for reporting this one! 🙏 Let us know if you run into any other issues and we'll be happy to help.
